octavia_f5/controller/worker/sync_manager.py

```diff
@@ -193,23 +193,29 @@ def tenant_delete(self, network_id, device=None):
         :return: True if success, else False
         """
         tenant = m_part.get_name(network_id)
-
         if CONF.f5_agent.dry_run:
             LOG.debug("Faking tenant_delete, tenant='%s', device='%s'", tenant, device)
             return True
 
-        ret = self.bigip(device).delete(tenants=[tenant])
+        # Instead of HTTP DELETE we POST an empty tenant declaration.
+        # It works just as well, but additionally leaves us control over declaration persistency.
+        response = self.tenant_update(network_id=network_id, device=device)
```

Comment:
Don't chain `tenant_delete` -> `tenant_update`, both have failover hooks and retry conditions, that would be doubled if triggered. This is bad style and unwanted behaviour. Use instead the self.bigip(device).post() function as used in tenant_update.